cmd/cogito/main_test.go

```diff
@@ -1,13 +1,113 @@
 package main
 
 import (
+	"bytes"
+	"io"
+	"strings"
 	"testing"
 
 	"gotest.tools/v3/assert"
 )
 
-func TestRunSmoke(t *testing.T) {
-	err := run(nil, nil, []string{"foo"})
+func TestRunSmokeSuccess(t *testing.T) {
+	type testCase struct {
+		name    string
+		args    []string
+		in      string
+		wantOut string
+	}
 
+	test := func(t *testing.T, tc testCase) {
+		in := strings.NewReader(tc.in)
+		var out bytes.Buffer
+
+		err := run(in, &out, io.Discard, tc.args)
+
+		assert.NilError(t, err)
+		have := out.String()
```

Comment:
Are you missing an empty line between L26 and L27?
(given your AAA comment?)

Reply:
If I'm not wrong, in this case they are in the same assert group.

Reply:
Arrange lines 21-22
blank line 23
Act line 24
blank line 25
Assert lines 26-28

Seems correct to me